src/js/components/Beacon/BeaconQueryUi.js

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useAppSelector } from '@/hooks';
+import { Button, Card, Form, Select, Tooltip } from 'antd';
+import { InfoCircleOutlined } from '@ant-design/icons';
+import Filters from './Filters';
+import { makeBeaconQuery } from '../../features/beacon/beaconQuery';
+import BeaconSearchResults from './BeaconSearchResults';
+import VariantsForm from './VariantsForm';
+
+// TODOs
+// form verification
+// example searches, either hardcoded or configurable
+// only render variants part of the form if variants are present
+// better helptext (#1691)
+
+const UI_INSTRUCTIONS = 'Search by genomic variants, clinical metadata or both.';
+const VARIANTS_HELP =
+  'Variants search requires the fields Chromosome, Variant start, Assembly ID, and at least one of Variant end or Alternate base(s).';
+const METADATA_HELP = 'Search over clinical or phenotypic properties.';
+const STARTER_FILTER = { index: 1, active: true };
+
+const WRAPPER_STYLE = {
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+};
+
+const VARIANTS_FORM_STYLE = {
+  maxWidth: '1200px',
+  display: 'flex',
+  flexWrap: 'wrap',
+};
+
+const CARD_BODY_STYLE = {
+  padding: '0 24px 5px 24px',
+};
+
+const CARD_HEAD_STYLE = {
+  border: '0',
+};
+
+const INNER_CARD_STYLE = {
+  margin: '0 5px 0 0',
+};
+
+const BUTTON_AREA_STYLE = {
+  padding: '20px 5px',
+};
+
+const BUTTON_STYLE = {
+  margin: '0 10px 0 0',
+};
+
+const BeaconQueryUi = () => {
+  const config = useSelector((state) => state?.beaconConfig?.config);
+  const beaconAssemblyIds = Object.keys(config?.overview?.counts?.variants ?? {});
+  const [filters, setFilters] = useState([STARTER_FILTER]);
+  const [form] = Form.useForm();
+  const querySections = useAppSelector((state) => state.query.querySections);
+  const beaconUrl = useAppSelector((state) => state.config?.beaconUrl);
+
+  const dispatch = useDispatch();
+  const launchEmptyQuery = () => dispatch(makeBeaconQuery(requestPayload({}, [])));
+
+  // get beacon stats on first render
+  useEffect(() => {
+    // wait for config
+    if (!beaconUrl) {
+      return;
+    }
+    launchEmptyQuery();
+  }, [config]);
+
+  // set assembly id options mataching what's in gohan
+  useEffect(() => {
+    form.setFieldsValue(formInitialValues);
+  }, [config]);
```

Comment:
cant the 2 be combined given the similar dependencies

Reply:
Yes, good catch. Judging from the comments, the first one started out with an empty dependency array, and I guess I changed it without noticing.